projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -218,8 +211,17 @@ public function get_item_schema() {
 	private function prepare_menu_item( array $menu_item ) {
 		global $submenu;
 
-		// Exclude unauthorized menu items.
-		if ( ! current_user_can( $menu_item[1] ) ) {
+		$current_user_can_access_menu = current_user_can( $menu_item[1] );
+		$has_first_menu_item          = isset( $submenu[ $menu_item[2] ][0] );
```

Comment:
Non-blocking: I know this is just a refactor keeping the existing behavior, but ideally we would show the parent menu as long as the user can access any of the submenus, not only the first one.

Reply:
I think we should look into how/where Core filters unauthorized submenus. From what I remember, in the endpoint, the submenus are not filtered yet based on their capabilities (I'll have to recheck) while in the Core menu, when they are displayed, I'm certain they are.

Reply:
> I think we should look into how/where Core filters unauthorized submenus

[All of them are filtered out](https://core.trac.wordpress.org/browser/trunk/src/wp-admin/menu-header.php?rev=49193#L201).

Reply:
Scratch that, my previous comment was referring to how the submenus are rendered, but my original comment was referring to how the parent menu is rendered, which always uses the first submenu ([but doesn't check its capability](https://github.com/Automattic/jetpack/pull/19604#discussion_r621274338)), so it's fine to leave this as-is, so we're aligned with Core.